packages/pools/src/router/__tests__/route.spec.ts

```diff
@@ -0,0 +1,109 @@
+import { Dec } from "@keplr-wallet/unit";
+
+import { calculateWeightForRoute, Route } from "../route";
+import { makeMockRoutablePool, MockRoutablePool } from "./pool";
+
+describe("route weight calculations", () => {
+  describe("calculateWeightForRoute", () => {
+    const lowLiqPools = [
+      makeMockRoutablePool("100", ["tokenA", "tokenB"]),
+      makeMockRoutablePool("50", ["tokenB", "tokenC"]),
+      makeMockRoutablePool("20", ["tokenC", "tokenD"]),
+    ];
+    const highLiqPools = [
+      makeMockRoutablePool("1000", ["tokenA", "tokenB"]),
+      makeMockRoutablePool("500", ["tokenB", "tokenC"]),
+      makeMockRoutablePool("200", ["tokenC", "tokenD"]),
+    ];
+    /** Higher liquidity at the end of the route */
+    const lowDepthPools = [
+      makeMockRoutablePool("20", ["tokenA", "tokenB"]),
+      makeMockRoutablePool("500", ["tokenB", "tokenC"]),
+      makeMockRoutablePool("800", ["tokenC", "tokenD"]),
+    ];
+
+    it("should return a higher weight for a route with higher liquidity", async () => {
+      const tokenInDenom = "tokenA";
+      const routeLowLiq: Route = {
+        pools: lowLiqPools,
+        tokenOutDenoms: getTokenOuts(lowLiqPools, tokenInDenom),
+        tokenInDenom,
+      };
+      const routeHighLiq: Route = {
+        pools: highLiqPools,
+        tokenOutDenoms: getTokenOuts(highLiqPools, tokenInDenom),
+        tokenInDenom,
+      };
+      const lowLiqWeight = await calculateWeightForRoute(
+        routeLowLiq,
+        makeTvlFunc(lowLiqPools)
+      );
+      const highLiqWeight = await calculateWeightForRoute(
+        routeHighLiq,
+        makeTvlFunc(highLiqPools)
+      );
+      expect(lowLiqWeight.lt(highLiqWeight)).toBeTruthy();
```

Comment:
this fails locally - should this return lower weight for route with higher liquidity instead since lower weight is better?